btree/btree.c

```diff
@@ -1853,105 +1853,25 @@ M0_INTERNAL void m0_btree_mod_fini(void)
-			lru_space_wm_high   = wm_high;
-		} else {
-			lru_space_wm_high   = LUSW_HIGH;
-		}
-	} else if (wm_low == -1 && wm_high == -1) {
-		lru_space_wm_low  = LUSW_LOW;
-		lru_space_wm_high = LUSW_HIGH;
-		if (wm_target < LUSW_HIGH &&
-		    (wm_target > LUSW_LOW || wm_target == LUSW_LOW)) {
-			lru_space_wm_target = wm_target;
-		} else {
-			lru_space_wm_target = LUSW_TARGET;
-		}
-	} else if (wm_target == -1 && wm_high == -1) {
-		lru_space_wm_target = LUSW_TARGET;
-		lru_space_wm_high   = LUSW_HIGH;
-		if (wm_low < LUSW_HIGH &&
-		    (wm_low < LUSW_TARGET || wm_low == LUSW_TARGET)) {
-			lru_space_wm_low    = wm_low;
-		} else {
-			lru_space_wm_low    = LUSW_LOW;
-		}
-	} else if (wm_low == -1) {
-		lru_space_wm_low = LUSW_LOW;
-		if ((wm_target > LUSW_LOW || wm_target == LUSW_LOW) &&
-		    wm_target < wm_high) {
-			lru_space_wm_target = wm_target;
-		} else {
-			lru_space_wm_target = LUSW_TARGET;
-		}
+	lrulist_trickle_release = (slow_lru_mem_release != 0) ? true : false;
 
-		if (wm_high > LUSW_LOW && wm_target < wm_high) {
-			lru_space_wm_high = wm_high;
-		} else {
-			lru_space_wm_high = LUSW_HIGH;
-		}
-	} else if (wm_target == -1) {
-		lru_space_wm_target = LUSW_TARGET;
-		if ((wm_low < LUSW_TARGET || wm_low == LUSW_TARGET) &&
-		    wm_low < wm_high) {
-			lru_space_wm_low = wm_low;
-		} else {
-			lru_space_wm_low = LUSW_LOW;
-		}
+	lru_space_wm_low = (wm_low == 0) ? LUSW_LOW : wm_low;
+	lru_space_wm_target = (wm_target == 0) ? LUSW_TARGET : wm_target;
+	lru_space_wm_high = (wm_high == 0) ? LUSW_HIGH : wm_high;
 
-		if (wm_high > wm_low && LUSW_TARGET < wm_high) {
-			lru_space_wm_high = wm_high;
-		} else {
-			lru_space_wm_high = LUSW_HIGH;
-		}
-	} else if (wm_high == -1) {
-		lru_space_wm_high   = LUSW_HIGH;
-		if((wm_low < wm_target || wm_low == wm_target) &&
-		   wm_low < LUSW_HIGH) {
-			lru_space_wm_low = wm_low;
-		} else {
-			lru_space_wm_low = LUSW_LOW;
-		}
+	M0_ASSERT(lru_space_wm_high >= lru_space_wm_target &&
+		  lru_space_wm_target >= lru_space_wm_low);
 
-		if ((wm_target > wm_low || wm_target == wm_low) &&
-		    wm_target < LUSW_HIGH) {
-			lru_space_wm_target = wm_target;
-		} else {
-			lru_space_wm_target = LUSW_TARGET;
-		}
-	} else {
-		if((wm_low < wm_target || wm_low == wm_target) &&
-		   wm_target < wm_high ) {
-			lru_space_wm_low    = wm_low;
-			lru_space_wm_target = wm_target;
-			lru_space_wm_high   = wm_high;
-		} else {
-			lru_space_wm_low    = LUSW_LOW;
-			lru_space_wm_target = LUSW_TARGET;
-			lru_space_wm_high   = LUSW_HIGH;
-		}
-	}
-	M0_LOG(M0_INFO, "Btree LRU List Watermarks: Low - %"PRIi64" Mid -"
+	M0_LOG(M0_DEBUG, "Btree LRU List Watermarks: Low - %"PRIi64" Mid - "
 	       "%"PRIi64" High - %"PRIi64" \n", lru_space_wm_low,
 	       lru_space_wm_target, lru_space_wm_high);
-	M0_LOG(M0_INFO, "Btree LRU List trickle release: %"PRIi64" \n",
-	       set_lrulist_trickle_release);
+	M0_LOG(M0_INFO, "Btree LRU List trickle release: %s \n",
```

Comment:
Use the level same as the level on line 1870.

Reply:
Done

Reply:
Sorry I meant it to be the other way - use M0_INFO for both these log messages. Since this call is done only once when the cluster boots it is not going to flood the trace file but we can use this info for debugging if possible,